pkg/api/api.go

```diff
@@ -65,17 +65,29 @@ type API struct {
 	authMiddleware middleware.Func
 	server         *server.Server
 	logger         log.Logger
+
+	querierRequestDuration *prometheus.HistogramVec
 }
 
-func New(cfg Config, s *server.Server, logger log.Logger) (*API, error) {
+func New(cfg Config, s *server.Server, logger log.Logger, reg prometheus.Registerer) (*API, error) {
 	// Ensure the encoded path is used. Required for the rules API
 	s.HTTP.UseEncodedPath()
 
+	// Prometheus histograms for requests.
+	querierRequestDuration := prometheus.NewHistogramVec(prometheus.HistogramOpts{
```

Comment:
Have you considered moving this to `RegisterQuerier()`? The metric would be registered only if `RegisterQuerier()` is used and the change isolated there.

Reply:
I wanted to avoid passing a `prometheus.Registerer` to that function. However, maybe it would make more sense to store the Registerer in the API struct and and initiate the metric is in the `RegisterQuerier` function.

Reply:
Yes, or you can pass the metric itself to the `RegisterQuerier()` (ie. in Thanos it's quite a common design pattern).

Reply:
👍 Cool I updated it to pass a metric as a parameter.